RobotCode/main.py

```diff
@@ -0,0 +1,247 @@
+            ### Shorten stuff / Print ball location
+            try:
+                ballX=processedData.balls[-1].x
+                ballY=processedData.balls[-1].y
+                print("Ball   X:",ballX," Y:",ballY)
+            except:
+                print("Cant see a ball")
+            
+            
+            ##### STATEMACHINE #####
+
+            if state == State.STOPPED:
+                print("stopped")
+                time.sleep(0.5) 
+                continue
+
+
+            ### SPINN UNTIL BOT SEES BALL ###
+            elif state == State.FINDBALL:
+                ### robot tries to find ball
+                try:
+                    if len(processedData.balls)>0:
+                        if ballX >= middle_x - find_ball_tolerance and ballX <= middle_x + find_ball_tolerance:
+                            ### robot found ball
+                            state = State.GETCLOSE
+                            print("findball->getclose")
+                    robot.findaball(processedData)
+
+                except:
+                    print("findball error")
+
+
+            ### GET CLOSE TO THE BALL ###
+            elif state == State.GETCLOSE:
+                try:
+                    if ballY > 250:
+                        try:
+                            if ibasket.distance < 700 and ibasket.distance>0:
+                                print("BASKET TOO CLOSE!!!!")
+                                robot.turn45()
+                            else:    
+                                print("getclose->orbit")
+                                state = State.ORBIT
+
+                        except:
+                            print("getclose->orbit")
+                            state = State.ORBIT
+                        
+                    elif len(processedData.balls) <= 0:
+                        print("getclose->findball")
+                        state = State.FINDBALL
+
+                    else:
+                        robot.getclose(ballX,ballY,middle_x)
+
+                except:
+                    print("getclose error")
+
+                if len(processedData.balls) <= 0:
+                    print("getclose->findball")
+                    state = State.FINDBALL
+
+
+            ### ORBIT AROUND THE BALL UNTILL BASKET CENTERED ###
+            elif state == State.ORBIT:
+                try:
+                    print("Orbiting")
+                    try:
+                        speed_x = -robot.controller(ibasket.x, middle_x, x_scale=1500, y_scale=(orbit_max_speed - 5))
+                    except:
+                        print("No basket")
+                        speed_x = orbit_max_speed
+
+                    if ibasket.x > middle_x - aim_tolerance and ibasket.x < middle_x + aim_tolerance:
+                        print("making shot")
+                        temptimer = 0
+                        state = State.MAKESHOT
+
+                    elif len(processedData.balls) <= 0 or ballY < 200:
```

Comment:
Use a ratio of screen resolution to be resolution agnostic

Reply:
Here you can have camera.rgb_height * 42. So when the ball is in the upper 58% of the screen, stuff happens